message/infrastructure/io/marshal.go

```diff
@@ -0,0 +1,98 @@
+package io
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ThreeDotsLabs/watermill"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// MarshalMessageFunc packages the message into a byte slice.
+// The topic argument is there because some writers (i.e. loggers) might want to present the topic as part of their output.
+type MarshalMessageFunc func(topic string, msg *message.Message) ([]byte, error)
+
+// PayloadMarshalFunc dumps the message's payload, discarding the remaining fields of the message.
+// The output is always terminated with EOL byte.
+//
+// This basic unmarshaler function may be used e.g. to write just the message payloads to stdout or to a file,
+// without cluttering the output with metadata and UUIDs.
+func PayloadMarshalFunc(topic string, msg *message.Message) ([]byte, error) {
+	return append(msg.Payload, '\n'), nil
+}
+
+// PrettyPayloadMarshalFunc dumps the message's payload.
```

Comment:
when I seen `Pretty` I expected to have formatted json or struct in "pretty format" - with newlines, tabs etc.

Reply:
b3ae9f252ccb6e90efcf9f72512765a9df71cff4 changed to a more explicit `TimestampTopicPayloadMarshalFunc`. Explicit is better than implicit :poultry_leg: